chain-signatures/node/src/indexer_eth.rs

```diff
@@ -920,87 +821,54 @@ async fn process_block(
-                }
-
-                tracing::warn!(
-                    "Block number {block_number} with hash: {block_hash:?} not in finalized epochs. "
-                );
+        else {
+            return Err(anyhow::anyhow!("Failed to receive indexed requests"));
+        };
 
-                // Check if block is in history (older than our oldest epoch)
-                if !finalized_epochs.is_empty() {
-                    let oldest_epoch = &finalized_epochs[0];
-                    if block_number < oldest_epoch.start_block {
-                        tracing::error!("Block {block_number} is in history");
-                        //TODO: handle the block in history case which happens when a block is retried
-                        break;
-                    }
-                }
+        // Wait for finalized block if needed
+        while finalized_block_number.is_none() || block_number > finalized_block_number.unwrap() {
+            let Some(new_finalized_block) = finalized_block_rx.recv().await else {
+                return Err(anyhow::anyhow!("Failed to receive finalized blocks"));
+            };
+            finalized_block_number.replace(new_finalized_block);
+        }
 
-                // Wait for new finalized epoch
-                let Some(received_epoch) = finalized_epoch_rx.recv().await else {
-                    tracing::warn!("Failed to receive finalized blocks");
-                    return Err(anyhow::anyhow!("Failed to receive finalized blocks"));
-                };
-
-                let start_block = received_epoch.start_block;
-                let end_block = received_epoch.end_block;
-                let blocks = received_epoch.blocks;
-
-                let new_epoch = FinalizedEpoch {
-                    start_block,
-                    end_block,
-                    blocks,
-                };
-
-                // Update epochs list, keeping only the last 2
-                finalized_epochs.push(new_epoch);
-                if finalized_epochs.len() > 2 {
-                    finalized_epochs.remove(0);
-                }
-            }
+        // Verify block hash and send requests
+        let block =
+            fetch_block_by_number(helios_client, block_number, 5, Duration::from_millis(200))
+                .await
+                .map_err(|e| {
+                    tracing::warn!("Failed to fetch block {block_number}: {e:?}");
+                    anyhow!("Failed to fetch block")
+                })?;
+
+        if block.header.hash == block_hash {
+            tracing::info!("Block {block_number} is finalized!");
+            send_indexed_requests(
+                indexed_requests,
+                sign_tx.clone(),
+                node_near_account_id.clone(),
+            );
+        } else {
+            tracing::error!(
+                "Block {block_number} hash mismatch: expected {block_hash:?}, got {:?}. Chain re-orged.",
+                block.header.hash
+            );
```

Comment:
Is there something we should be doing here in the else portion? If something else is doing work after the fact to resolve the reorg, we should have a comment stating what's going to happen